### PR TITLE
PDCT 395/id generation collections

### PR DESCRIPTION
**Pull request by diversemix**
# Description

Changes for collections 
- Have import_id generated
- PUT endpoint now takes the import_id
- Updates tests
- 
## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

## How Has This Been Tested?

Please describe the tests that you added to verify your changes.

## Reviewer Checklist

- [ ] The PR represents a single feature (small driveby fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [ ] The code is sufficiently commented, particularly in hard-to-understand areas
- [ ] Any required documentation updates have been made
- [ ] Any TODOs added are captured in future tickets
- [ ] No FIXMEs remain

### CODE REVIEW

File: app/model/collection.py [hunk 1]
```diff
@@ -14,7 +14,15 @@ class CollectionReadDTO(BaseModel):
...
     description: str
     organisation: str
+
+
+class CollectionCreateDTO(BaseModel):
+    """Representation of a Collection for creating."""
+
+    title: str
+    description: str
+    # families: list[str] TODO: Ask Patrick if we want this as an option?
```
Comment (katybaulch) [inline app/model/collection.py:27]: Have we confirmed with Patrick whether or not we want this option?

File: app/service/collection.py [hunk 2]
```diff
@@ -80,46 +92,53 @@ def validate_import_id(import_id: str) -> None:
...
 
+    :param import_id str: The collection import_id to change.
     :param CollectionDTO collection: The DTO with all the values to change (or keep).
     :raises RepositoryError: raised on a database error.
     :raises ValidationError: raised should the import_id be invalid.
     :return Optional[CollectionDTO]: The updated collection or None if not updated.
     """
-    validate_import_id(collection.import_id)
 
     # TODO: implement changing of a collection's organisation
```
Comment (katybaulch) [inline app/service/collection.py:107]: Do we still need this TODO?

File: app/service/document.py [hunk 3]
```diff
@@ -97,9 +97,14 @@ def update(
     # TODO: implement changing of a document's organisation
```
Comment (katybaulch) [inline app/service/document.py:97]: Do we still need this TODO?

**Review by katybaulch — APPROVED** [review]
Looks good to me!

File: app/model/collection.py (hunk 1 above)
Comment (diversemix) [reply app/model/collection.py:27]: Not yet ... but I tend to litter TODO's around to keep velocity - keeping a task before release that we go through and remove / ticket / fix them :)

File: app/service/collection.py (hunk 2 above)
Comment (diversemix) [reply app/service/collection.py:107]: I think so - as we may not want to do that at all

File: app/service/document.py (hunk 3 above)
Comment (diversemix) [reply app/service/document.py:97]: Yeah - good to leave them in to process them all at once

**Pull request merged**